rclcpp/include/rclcpp/time_utils.hpp

```diff
@@ -0,0 +1,74 @@
+// Copyright 2019 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCLCPP__TIME_UTILS_HPP_
+#define RCLCPP__TIME_UTILS_HPP_
+
+#include <stdexcept>
+#include <type_traits>
+
+#include "rcl/time.h"
+
+namespace rclcpp
+{
+
+/** \defgroup time_conversions Group contains conversion helper functions
+ *  @{
+ */
+
+/// Combines signed seconds and unsigned nanoseconds into one field
+/**
+ * Given seconds and nanoseconds the function merges both into one field.
+ *
+ * Pseudo-code:
+ * output = sec * 1e10 + nanoseconds % 1e10
+ *
+ * \param sec seconds
+ * \param nano nanoseconds
+ * \return field containing seconds and nanoseconds
+ */
+inline int64_t join_to_nano(int32_t sec, uint32_t nano) noexcept
+{
+  static constexpr uint32_t max_nano = 1000000000;
+  // store seconds and nanoseconds in one field
+  const auto only_nano = static_cast<int64_t>(nano % max_nano);
+  return RCL_S_TO_NS(static_cast<int64_t>(sec)) + only_nano;
+}
+
+/// Split signed data into seconds and nanoseconds
+/**
+ * Given a full field of nanoseconds the function splits it up into separate
+ * fields.
+ *
+ * Pseudo-code:
+ * sec = full / 1e10
+ * nano = full % 1e10
+ *
+ * \param[in] full contains both, seconds and nanoseconds
+ * \param[out] sec contains only seconds
+ * \param[out] nano contains only nanoseconds
+ */
+inline void split_from_nano(int64_t full, int32_t & sec, uint32_t & nano) noexcept
```

Comment:
I feel like this function should be a new macro in here: https://github.com/ros2/rcutils/blob/master/include/rcutils/time.h

Reply:
same as above